baseplate/context/redis.py

```diff
@@ -4,55 +4,83 @@
 from __future__ import unicode_literals
 
 import redis
+import redis.client
 
 from . import ContextFactory
 
 
 class RedisContextFactory(ContextFactory):
     """Redis client context factory.
 
-    This factory will attach a proxy object which acts like a
-    :py:class:`redis.StrictRedis` object to an attribute on the :term:`context
-    object`. When redis commands are executed via this proxy object, they will
-    use connections from the provided :py:class:`redis.ConnectionPool` and
-    automatically record diagnostic information.
-
-    .. note::
-
-        Pipelines, transactions, locks, and pubsub are currently unsupported.
+    This factory will attach a
+    :py:class:`~baseplate.context.redis.MonitoredRedisConnection` to an
+    attribute on the :term:`context object`. When redis commands are executed
+    via this connection object, they will use connections from the provided
+    :py:class:`redis.ConnectionPool` and automatically record diagnostic
+    information.
 
     :param redis.ConnectionPool connection_pool: A connection pool.
 
+    :returns: :py:class:`~baseplate.context.redis.MonitoredRedisConnection`
+
     """
     def __init__(self, connection_pool):
         self.connection_pool = connection_pool
 
     def make_object_for_context(self, name, root_span):
-        return RedisConnectionAdapter(name, root_span, self.connection_pool)
+        return MonitoredRedisConnection(name, root_span, self.connection_pool)
+
+
+class MonitoredRedisConnection(redis.StrictRedis):
+    """Redis connection that collects diagnostic information.
 
+    This connection acts like :py:class:`redis.StrictRedis` except that all
+    operations are automatically wrapped with diagnostic collection.
+
+    The interface is the same as that class except for the
+    :py:meth:`~baseplate.context.redis.MonitoredRedisConnection.pipeline`
+    method.
+
+    .. note:: Locks and pubsub are currently unsupported.
```

Comment:
According to your other note transactions are still unsupported right? Only in pipelines?